features/include/pcl/features/impl/good.hpp

```diff
@@ -0,0 +1,750 @@
+
+template <typename PointInT>
+void 
+pcl::GOODEstimation<PointInT>::setInputCloud (boost::shared_ptr<pcl::PointCloud<PointInT> > cloud)
+{ 
+  input_ = cloud; 
+};
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::setNumberOfBins (unsigned int number_of_bins)
+{
+  number_of_bins_ = number_of_bins;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::setThreshold (float threshold)
+{
+  threshold_ = threshold;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::getOrthographicProjections (std::vector < boost::shared_ptr<pcl::PointCloud<PointInT> > > &vector_of_projected_views)
+{
+  vector_of_projected_views = vector_of_projected_views_;  
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::getTransformedObject (boost::shared_ptr<pcl::PointCloud<PointInT> > &transformed_point_cloud)
+{
+  transformed_point_cloud = transformed_point_cloud_;
+} 
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void
+pcl::GOODEstimation<PointInT>::getCenterOfObjectBoundingBox (pcl::PointXYZ &center_of_bbox)
+{
+  center_of_bbox = center_of_bbox_;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>     
+void
+pcl::GOODEstimation<PointInT>::getObjectBoundingBoxDimensions(pcl::PointXYZ &bbox_dimensions)
+{
+  bbox_dimensions = bbox_dimensions_;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>  
+void 
+pcl::GOODEstimation<PointInT>::getOrderOfProjectedPlanes(std::string &order_of_projected_plane)
+{
+  order_of_projected_plane = order_of_projected_plane_;
+}
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void 
+pcl::GOODEstimation<PointInT>::getTransformationMatrix( Eigen::Matrix4f &transformation)
+{
+  transformation = transformation_;
+} 
+//////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename PointInT>
+void 
+pcl::GOODEstimation<PointInT>::computeBoundingBoxDimensions (boost::shared_ptr<pcl::PointCloud<PointInT> > pc, pcl::PointXYZ& dimensions)
+{
+  PointInT minimum_pt;
+  PointInT maximum_pt;
+  pcl::getMinMax3D(*pc, minimum_pt, maximum_pt); // min max for bounding box
+  dimensions.x = (maximum_pt.x - minimum_pt.x); 
+  dimensions.y = (maximum_pt.y - minimum_pt.y); 
+  dimensions.z = (maximum_pt.z - minimum_pt.z); 	
+}
```

Comment:
Fair enough. What you need is `maximum_pt.getVector3fMap () - minimum_pt.getVector3fMap ()`.